runtime-modules/forum/src/lib.rs

```diff
@@ -746,6 +752,26 @@ decl_module! {
             Ok(())
         }
 
+        fn delete_category(origin, moderator_id: T::ModeratorId, category_id: T::CategoryId) -> dispatch::Result {
+            // Ensure data migration is done
+            Self::ensure_data_migration_done()?;
+
+            Self::ensure_can_delete_category(origin, &moderator_id, &category_id)?;
+
+            let category = <CategoryById<T>>::get(category_id);
```

Comment:
This is not as safe as it can be. If `ensure_can_delete_category` has been successful, then we should be able to statically guarantee that the category instance exist, rather than looking it up again in the map, which can fail at runtime.

Reply:
Redudant get was removed.